R/rdeephaven/src/client.cpp

```diff
@@ -14,17 +15,257 @@
+    return new AggregateWrapper(deephaven::client::Aggregate::avg(columnSpecs));
+}
+
+AggregateWrapper* INTERNAL_wAvg(std::string weightColumn, std::vector<std::string> columnSpecs) {
+    return new AggregateWrapper(deephaven::client::Aggregate::wavg(weightColumn, columnSpecs));
+}
+
+AggregateWrapper* INTERNAL_median(std::vector<std::string> columnSpecs) {
+    return new AggregateWrapper(deephaven::client::Aggregate::med(columnSpecs));
+}
+
+AggregateWrapper* INTERNAL_var(std::vector<std::string> columnSpecs) {
+    return new AggregateWrapper(deephaven::client::Aggregate::var(columnSpecs));
+}
+
+AggregateWrapper* INTERNAL_std(std::vector<std::string> columnSpecs) {
+    return new AggregateWrapper(deephaven::client::Aggregate::std(columnSpecs));
+}
+
+AggregateWrapper* INTERNAL_percentile(double percentile, std::vector<std::string> columnSpecs) {
+    return new AggregateWrapper(deephaven::client::Aggregate::pct(percentile, false, columnSpecs));
+}
+
+AggregateWrapper* INTERNAL_count(std::string columnSpec) {
+    return new AggregateWrapper(deephaven::client::Aggregate::count(columnSpec));
+}
+
+
 class TableHandleWrapper {
 public:
     TableHandleWrapper(deephaven::client::TableHandle ref_table) :
         internal_tbl_hdl(std::move(ref_table)) {};
 
-    // TODO: DEEPHAVEN QUERY METHODS WILL GO HERE
+    // TABLE OPERATIONS
+
+    // FILTER OPERATIONS
+
+    TableHandleWrapper* select(std::vector<std::string> columnSpecs) {
+        return new TableHandleWrapper(internal_tbl_hdl.select(columnSpecs));
+    };
+
+    TableHandleWrapper* view(std::vector<std::string> columnSpecs) {
+        return new TableHandleWrapper(internal_tbl_hdl.view(columnSpecs));
+    };
+
+    TableHandleWrapper* update(std::vector<std::string> columnSpecs) {
+        return new TableHandleWrapper(internal_tbl_hdl.update(columnSpecs));
+    };
+
+    TableHandleWrapper* updateView(std::vector<std::string> columnSpecs) {
+        return new TableHandleWrapper(internal_tbl_hdl.updateView(columnSpecs));
+    };
+
+    TableHandleWrapper* dropColumns(std::vector<std::string> columnSpecs) {
+        return new TableHandleWrapper(internal_tbl_hdl.dropColumns(columnSpecs));
+    };
+
+    TableHandleWrapper* where(std::string condition) {
+        return new TableHandleWrapper(internal_tbl_hdl.where(condition));
+    };
+
+    TableHandleWrapper* groupBy(std::vector<std::string> columnSpecs) {
+        return new TableHandleWrapper(internal_tbl_hdl.by(columnSpecs));
+    };
+
+    TableHandleWrapper* ungroup(std::vector<std::string> groupByColumns) {
+        return new TableHandleWrapper(internal_tbl_hdl.ungroup(false, groupByColumns));
+    };
+
+    // AGGREGATION OPERATIONS
+
+    TableHandleWrapper* aggBy(Rcpp::List aggregations, std::vector<std::string> groupByColumns) {
+        std::vector<deephaven::client::Aggregate> converted_aggregations = convertRcppListToVectorOfTypeAggregate(aggregations);
```

Comment:
Not a fan, I like to see everything that's happening in maximum detail because I don't understand this well enough for hiding explicit types to be helpful